plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+				for _, v := range h.Metrics {
+					switch v {
+					case "jvm":
+						h.getJVMStatistics(acc, serverURL, execAsDomain, host, server)
+					case "web":
+						if isEAP7 {
+							h.getUndertowStatistics(acc, serverURL, execAsDomain, host, server, "ajp")
+							h.getUndertowStatistics(acc, serverURL, execAsDomain, host, server, "http")
+							h.getUndertowStatistics(acc, serverURL, execAsDomain, host, server, "https")
+						} else {
+							h.getWebStatistics(acc, serverURL, execAsDomain, host, server, "ajp")
+							h.getWebStatistics(acc, serverURL, execAsDomain, host, server, "http")
+						}
+					case "deployment":
+						h.getServerDeploymentStatistics(acc, serverURL, execAsDomain, host, server)
+					case "database":
+						h.getDatasourceStatistics(acc, serverURL, execAsDomain, host, server)
+					case "jms":
+						if isEAP7 {
+							h.getJMSStatistics(acc, serverURL, execAsDomain, host, server, "messaging-activemq", GetJMSQueueStat)
+							h.getJMSStatistics(acc, serverURL, execAsDomain, host, server, "messaging-activemq", GetJMSTopicStat)
+						} else {
+							h.getJMSStatistics(acc, serverURL, execAsDomain, host, server, "messaging", GetJMSQueueStat)
+							h.getJMSStatistics(acc, serverURL, execAsDomain, host, server, "messaging", GetJMSTopicStat)
+						}
+					case "transaction":
+						h.getTransactionStatistics(acc, serverURL, execAsDomain, host, server)
+					default:
+						log.Printf("E! Jboss doesn't exist the metric set %s\n", v)
+					}
+				}
+			}
+		}(host)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+// Gathers Transaction data from a particular host
+// Parameters:
+//     acc      : The telegraf Accumulator to use
+//     serverURL: endpoint to send request to
+//     host     : the host being queried
+//     server   : the server being queried
+//
+// Returns:
+//     error: Any error that may have occurred
+
+func (h *JBoss) getTransactionStatistics(
+	acc telegraf.Accumulator,
+	serverURL string,
+	execAsDomain bool,
+	host string,
+	serverName string,
+) error {
+	adr := OrderedMap{}
+
+	if execAsDomain {
+		adr = OrderedMap{
+			{"host", host},
+			{"server", serverName},
+			{"subsystem", "transactions"},
+		}
+	} else {
+		adr = OrderedMap{
+			{"subsystem", "transactions"},
+		}
+	}
+
+	bodyContent, err := h.createRequestBody(GetTransactionStat, adr)
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	}
+
+	out, err := h.doRequest(serverURL, bodyContent)
+
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	} else {
+		server := TransactionResponse{}
+		if err = json.Unmarshal(out, &server); err != nil {
+			return fmt.Errorf("Error decoding JSON response: %s : %s", out, err)
+		}
+
+		fields := make(map[string]interface{})
+		for key, value := range server.Result {
+			if strings.Contains(key, "number-of") {
+				fields[key], _ = strconv.ParseInt(value.(string), 10, 64)
```

Comment:
Please check the type assertion here! If, for whatever reason, `value` is not a `string` you will tear-down telegraf completely with a panic!